### PR TITLE
Added methods to send Images and CodeSnippets to users and channels

### PR DESCRIPTION
**Pull request by Geanik**
i've added a method to send an Java Image to a user (with a title)

and another to send a CodeSnippet
here you can define the title, the snippet itself and the language
with the information about the language the so called "fileType" attribute can be added, and slack can highlight that code
https://api.slack.com/types/file#file_types

### CODE REVIEW

**Comment by sshiman** [comment]
@Geanik Any plans to merge this into the master/new version?

**Comment by Geanik** [comment]
i don't know
haven't heard anything from the maintainers since i've opened this MR

**Comment by sshiman** [comment]
> i don't know
> haven't heard anything from the maintainers since i've opened this MR

@bcorne  would you be able to provide the ETA for the release of it (and a new version)?

**Comment by sshiman** [comment]
> i don't know
> haven't heard anything from the maintainers since i've opened this MR

@Geanik could it be becasue of "coverage" task has "failed" after build? I have contacted @bcorne on ETA

**Comment by sshiman** [comment]
@bcorne Do you still support this repository? would you like to pass rights to other team members by the chance if you can not approve yourself?

File: sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java [hunk 1]
```diff
@@ -645,6 +642,46 @@ private void stopConnectionMonitoring() {
...
+    }
+
+    public SlackMessageHandle<SlackMessageReply> sendCodeSnippet(SlackChannel channel, String title, String snippet, String language) {
+        SlackMessageHandle<SlackMessageReply> handle = new SlackMessageHandle<>(getNextMessageId());
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("token", authToken);
+        arguments.put("channels", channel.getId());
+        arguments.put("title", title);
+        arguments.put("filetype", language);
+        postSlackCommandWithFile(arguments, snippet.getBytes(), title, FILE_UPLOAD_COMMAND, handle);
```
Comment (gluck) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: using title as upload file name is ok ? (e.g. any restrictions in characters usable ?)

File: sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java [hunk 2]
```diff
@@ -645,6 +642,46 @@ private void stopConnectionMonitoring() {
...
+
+    public SlackMessageHandle<SlackMessageReply> sendImage(SlackChannel channel, String title, File img) {
+        SlackMessageHandle<SlackMessageReply> handle = new SlackMessageHandle<>(getNextMessageId());
+        Map<String, String> arguments = new HashMap<>();
+        arguments.put("token", authToken);
+        arguments.put("channels", channel.getId());
+        arguments.put("title", title);
+
+        // convert file to byte []
+        byte[] imgBytes = new byte[(int) img.length()];
```
Comment (gluck) [inline sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: would ideally use `addBinaryBody(... File ...)` overload to stream file to socket and avoid loading it up in memory (hopefully that's what this overload does), but no big deal

Comment (sshiman) [reply sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: @Geanik what do you think about file name question? we can generate GUID-like file name if @gluck think it's better like that?

File: sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java (hunk 1 above)
Comment (sshiman) [reply sources/src/main/java/com/ullink/slack/simpleslackapi/impl/SlackWebSocketSessionImpl.java]: @Geanik what do you think about file name question? we can generate GUID-like file name if @gluck think it's better like that?

**Pull request closed without merging**